File: Trabajo_7/JAVA/Ejercicios/Ejercicio_1.java
package Trabajo_7.Ejercicios;

class Vehiculo {
    private String marca;
    private String modelo;
    private int anio;
    private float precio;

    public Vehiculo(String marca, String modelo, int anio, float precio) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precio = precio;
    }

    public String getMarca() {
        return marca;
    }

    public String getModelo() {
        return modelo;
    }

    public int getAnio() {
        return anio;
    }

    public float getPrecio() {
        return precio;
    }
}


}

class Automovil extends Vehiculo {
    private int n_puertas;
    private String tipo_c;

    public Automovil(String marca, String modelo, int anio, float precio, int n_puertas, String tipo_c) {
        super(marca, modelo, anio, precio);
        this.n_puertas = n_puertas;
        this.tipo_c = tipo_c;
    }

    @Override
    public String toString() {
        return "Caracteristicas de Auto xd:\n" +
                "Marca            -> " + getMarca() + "\n" +
                "Modelo           -> " + getModelo() + "\n" +
                "Anio             -> " + getAnio() + "\n" +
                "Precio           -> " + getPrecio() + "\n" +
                "Numero Puertas   -> " + n_puertas + "\n" +
                "Tipo Combustible -> " + tipo_c;
    }
}

class Moto extends Vehiculo {
    private int cilindrada;

    public Moto(String marca, String modelo, int anio, float precio, int cilindrada) {
        super(marca, modelo, anio, precio);
        this.cilindrada = cilindrada;
    }

    @Override
    public String toString() {
        return "Caracteristicas de Moto:\n" +
                "Marca        -> " + getMarca() + "\n" +
                "Modelo       -> " + getModelo() + "\n" +
                "Anio         -> " + getAnio() + "\n" +
                "Precio       -> " + getPrecio() + "\n" +
                "Cilindrada   -> " + cilindrada + "cc";
    }
}

public class Ejercicio_1 {
    public static void main(String[] args) {
        Automovil auto_01 = new Automovil("Nissan", "WAA", 2005, 250000, 4, "Premium");
        Moto mototaxi_01 = new Moto("La", "Moto", 2010, 10000, 150);
        System.out.println(auto_01);
        System.out.println();
        System.out.println(mototaxi_01);
    }
}

File: Trabajo_7/CPP/Ejercicios/Ejercicio_1.cpp
#include <iostream>
using namespace std;

class Vehiculo{
private:
    string marca;
    string modelo;
    int anio;
    float precio;

public:
    Vehiculo(string brand, string model, int year, float price){
        this -> marca = brand;
        this -> modelo = model;
        this -> anio = year;
        this -> precio = price;
    }

    //Metodos G y S
    string getMarca() {
        return marca;
    }

    string getModelo() {
        return modelo;
    }

    int getAnio() {
        return anio;
    }

    float getPrecio() {
        return precio;
    }

};

class Automovil : public Vehiculo{
private:
    int n_puertas;
    string tipo_c;

public:
    Automovil(string brand, string model, int year, float price, int n_doors, string type_c)
            : Vehiculo(brand,model,year,price){
        this -> n_puertas = n_doors;
        this -> tipo_c = type_c;
    }

    friend ostream& operator << (ostream& os, Automovil Autito){
        os << "Caracteristicas de Auto xd:" << endl;
        os << "Marca            -> " << Autito.getMarca() << endl;
        os << "Modelo           -> " << Autito.getModelo() << endl;
        os << "Anio             -> " << Autito.getAnio() << endl;
        os << "Precio           -> " << Autito.getPrecio() << endl;
        os << "Numero Puertas   -> " << Autito.n_puertas << endl;
        os << "Tipo Combustible -> " << Autito.tipo_c << endl;
        return os;
    }

};

class Moto : public Vehiculo{
private:
    int cilindrada;

public:
    Moto(string brand, string model, int year, float price, int cilin)
            : Vehiculo(brand,model,year,price){
        this -> cilindrada = cilin;
    }

    friend ostream& operator << (ostream& os, Moto Motito){
        os << "Caracteristicas de Moto:" << endl;
        os << "Marca        -> " << Motito.getMarca() << endl;
        os << "Modelo       -> " << Motito.getModelo() << endl;
        os << "Anio         -> " << Motito.getAnio() << endl;
        os << "Precio       -> " << Motito.getPrecio() << endl;
        os << "Cilindrada   -> " << Motito.cilindrada << "cc" << endl;
        return os;
    }
};




int main(){
    Automovil Auto_01("Nissan", "WAA", 2005, 250000, 4, "Premium");
    Moto Mototaxi_01("La", "Moto", 2010, 10000, 150);
    cout << Auto_01;
    cout << "" << endl;
    cout << Mototaxi_01;
    return 0;
}